From f19b006b6753abb07e74c5f727066941516ece19 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Tue, 19 Jun 2018 09:39:47 +0200
Subject: [PATCH 1/2] sparse_io.c: Fix format string

Without this patch a warning is emitted by GCC:

warning: format '%lu' expects argument of type 'long unsigned int', but
         argument 3 has type 'uint64_t {aka long long unsigned int}' [-Wformat=]
   fprintf(stderr, "failed to alloc %lu\n", num * block_size);

This patches address this warning.
---
 lib/ext2fs/sparse_io.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/ext2fs/sparse_io.c b/lib/ext2fs/sparse_io.c
index a83baf1f..2f0ddc7f 100644
--- a/lib/ext2fs/sparse_io.c
+++ b/lib/ext2fs/sparse_io.c
@@ -4,6 +4,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdint.h>
+#include <inttypes.h>
 #include "ext2_fs.h"
 #include "ext2fs.h"
 
@@ -270,7 +271,7 @@ static errcode_t sparse_merge_blocks(struct sparse_map *sm, uint64_t start,
 
 	buf = calloc(num, block_size);
 	if (!buf) {
-		fprintf(stderr, "failed to alloc %lu\n", num * block_size);
+		fprintf(stderr, "failed to alloc %"PRIu64"\n", num * block_size);
 		return EXT2_ET_NO_MEMORY;
 	}
 
-- 
2.17.1

